### DISCUSSION TITLE
Show disk module only if mounted?

### DISCUSSION THREAD

**Discussion by rieje** [Alexays/Waybar#2052] [Q&A]
I have a disk module for the path `/data` which is where a media drive is mounted to. When it gets unmounted, the module shows the same info as my other disk module for `/home/user` (my system disk).

Is there a way to only show the disk module if the filesystem is mounted by the media disk?

Also was curious what is used to check disk size--I currently have interval set to 30 seconds but it would be nice to get close to real time--would like to know if this is relatively inexpensive to do. From what I understand, there's no event for determining disk size so polling is expensive, right?

**Comment by ethze** [comment]
Yes, it's possible to modify the disk module to show only when the filesystem is mounted. In Waybar, you can check if a specific mount point is active using shell commands in the configuration. You could create a script or use a condition to check whether the `/data` mount point exists, and then dynamically display the module.

Here’s an example of how you could configure it in Waybar:

```json
"modules-right": [
  "custom/media_disk"
],
"custom/media_disk": {
  "exec": "if mountpoint -q /data; then echo 'Disk Mounted'; else echo ''; fi",
  "interval": 30
}
